### PR TITLE
thingspeak service and example Fixes #1124

### LINKED ISSUE #1124: thingsspeak IOT api
https://thingspeak.com/channels/107204/feed.json
https://thingspeak.com/channels/public
https://www.mathworks.com/help/thingspeak/rest-api.html
- create / get charts: https://www.mathworks.com/help/thingspeak/create-a-chart.html

### PR DESCRIPTION
**Pull request by hamidzr**
continuing the work from #1126. 
- pagination support for searches
- added func to search by both tag and location at the same time ( no support from the provider )
- more details for channelDetails method
- new example
- light search result validation

blocked by #1344 #1341 
* this is needed for the upcoming eclipse experiment.

### CODE REVIEW

File: src/server/rpc/procedures/thingspeak-iot/thingspeak-iot.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+                longitude: longitude,
+                distance: distance === '' ? 100 : distance
+            })
+    };
+    limit = limit || 15;
+    return this._paginatedQueryOpts(queryOptions, limit).then(queryOptsList => {
+        return this._sendStruct(queryOptsList, searchParser);
+    });};
+
+thingspeakIoT.searchByBoth= function(tag, latitude, longitude, distance) {
```
Comment (brollb) [inline src/server/rpc/procedures/thingspeak-iot/thingspeak-iot.js]: This method name needs to be changed

**Comment by brollb** [comment]
Why does this have edits to static-map?

**Comment by brollb** [comment]
The example also doesn't seem to open... Did you export it as a role or the entire room?

**Comment by hamidzr** [comment]
the change to staticmap was a necessary fix which was addressed in #1345 and #1343. I was expecting it to be mergeable.

**Pull request merged**